src/main/java/com/mtab/spring/boot/controller/BkStrController.java

```diff
@@ -0,0 +1,43 @@
+package com.mtab.spring.boot.controller;
+
+import com.mtab.spring.boot.service.BkStrService;
+import com.mtab.spring.boot.service.models.Book;
+import com.mtab.spring.boot.service.models.BookPurchase;
+import com.mtab.spring.boot.service.models.NewPrice;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+public class BkStrController {
+    private final BkStrService bkStrService;
+
+    @Autowired
+    BkStrController(BkStrService bkStrService) {
+        this.bkStrService = bkStrService;
+    }
+
+    @PostMapping("/addBook")
```

Comment:
In RESTful API design you try not to put words like "get", "add", "delete" etc in the URL, preferring instead to let the HTTP method provide the context.

So I would have this be a PUT and the path be simply "book".